### PR TITLE
Simulation `eval_coords` bug 

### PR DESCRIPTION
**Pull request by j-c-c**
This PR addresses the fact that Cov3d clustering indices maybe permuted from the Simulation volume indices or `states`, mentioned in #823

### CODE REVIEW

File: src/aspire/source/simulation.py [hunk 1]
```diff
@@ -414,10 +414,13 @@ def eval_clustering(self, vol_idx):
         assert (
             len(vol_idx) == self.n
         ), f"Need {self.n} vol indexes to evaluate clustering"
-        # Remember that `states` is 1-indexed while vol_idx is 0-indexed
-        correctly_classified = np.sum(self.states - 1 == vol_idx)
+        # Remember that `states` is 1-indexed while vol_idx is 0-indexed.
+        # In addition, vol_idx may be improperly labeled so we check for
+        # the closest to 1-to-1 mapping between vol_idx and states.
+        remapped_vol_idx = (vol_idx + np.arange(self.C)[:, None]) % self.C
```
Comment (garrettwrong) [inline src/aspire/source/simulation.py]: Are you sure this is all the possible mappings... I'm not 100% but I think you may be forgetting `(self.C-1)!` of them.

**Review by garrettwrong — COMMENTED** [review]
This problem is famous.  For a statistic see [adjusted rand index](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.adjusted_rand_score.html).

For similar loosely related algorithms, see both `Edit Distance` family, the `Hungarian Problem` family, and bipartite matching. Although I think the statistic I mention is the exact thing to use if we are trying to solve this for the general case.

Let me know if the statistic will cover the issue, thanks.

File: src/aspire/source/simulation.py (hunk 1 above)
Comment (j-c-c) [reply src/aspire/source/simulation.py]: I'm definitely forgetting `(self.C - 1)!` mappings!

**Review by garrettwrong — DISMISSED** [review]
Code lgtm, thanks.

Can you squash this down to one commit? While doing that, you can add the `Co-authored-by:` on the commit so I get that elusive Pair Extraordinaire badge 😇 .

**Comment by j-c-c** [comment]
> Code lgtm, thanks.
> 
> Can you squash this down to one commit? While doing that, you can add the `Co-authored-by:` on the commit so I get that elusive Pair Extraordinaire badge 😇 .

Will do! Sorry forgot to add you to that!

**Review by garrettwrong — APPROVED** [review]

**Comment by garrettwrong** [comment]
I think we can spare Joakim on reviewing this one if you want to merge it in (we talked about it yesterday). Ty!

**Pull request merged**